core/tls-websocket-server/src/tls_common.rs

```diff
@@ -41,42 +35,26 @@ pub fn make_config(cert: &str, key: &str) -> WebSocketResult<Arc<rustls::ServerC
 	Ok(Arc::new(config))
 }
 
-fn load_certs(filename: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
-	let certfile = fs::File::open(filename).map_err(|e| {
-		WebSocketError::InvalidCertificate(format!(
-			"Failed to load certificate from file ({}): {:?}",
-			filename, e
-		))
-	})?;
-	let mut reader = BufReader::new(certfile);
-	rustls::internal::pemfile::certs(&mut reader).map_err(|_| {
-		WebSocketError::InvalidCertificate("Failed to parse certificate file".to_string())
-	})
+fn load_certs(content: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
+	let mut reader = BufReader::new(content.as_bytes());
+	rustls::internal::pemfile::certs(&mut reader)
+		.map_err(|_| WebSocketError::InvalidCertificate("Failed to parse certificate".to_string()))
 }
 
-fn load_private_key(filename: &str) -> WebSocketResult<rustls::PrivateKey> {
+fn load_private_key(content: &str) -> WebSocketResult<rustls::PrivateKey> {
 	let rsa_keys = {
-		let keyfile = fs::File::open(filename).map_err(|e| {
-			WebSocketError::InvalidPrivateKey(format!(
-				"Failed to load private key from file ({}): {:?}",
-				filename, e
-			))
-		})?;
-		let mut reader = BufReader::new(keyfile);
+		let mut reader = BufReader::new(content.as_bytes());
 
 		rustls::internal::pemfile::rsa_private_keys(&mut reader).map_err(|_| {
-			WebSocketError::InvalidPrivateKey("file contains invalid rsa private key".to_string())
+			WebSocketError::InvalidPrivateKey("Failed to parse RSA private key".to_string())
 		})?
 	};
 
 	let pkcs8_keys = {
-		let keyfile = fs::File::open(filename)
-			.map_err(|e| WebSocketError::InvalidPrivateKey(format!("{:?}", e)))?;
-		let mut reader = BufReader::new(keyfile);
+		let mut reader = BufReader::new(content.as_bytes());
```

Comment:
So the reason we generated these file before, was probably that we weren't aware of the fact that a `BufReader` can also take content directly, rather than a file name?

Reply:
Yes, I think so, that is the most likely reason.
(IMHO this is not intuitive, I was kind of surprised at first too)